presto-main/src/test/java/com/facebook/presto/sql/query/TestFilteredAggregations.java

```diff
@@ -86,4 +86,29 @@ public void testGroupingSets()
                         "(2, BIGINT '4', BIGINT '1'), " +
                         "(CAST(NULL AS INTEGER), BIGINT '5', BIGINT '2')");
     }
+
```

Comment:
With the PR

``presto:sf1> explain select sum(a + 1) filter (where a + 1 > 0) from (values (1), (2), (3)) t(a);
                                             Query Plan                         
--------------------------------------------------------------------------------
 - Output[_col0] => [sum:bigint]                                                
         Cost: {rows: 1 (9B), cpu: 111.00, memory: 18.00, network: 0.00}        
         _col0 := sum                                                           
     - Aggregate(FINAL) => [sum:bigint]                                         
             Cost: {rows: 1 (9B), cpu: 111.00, memory: 18.00, network: 0.00}    
             sum := "sum"("sum_9")                                              
         - LocalExchange[SINGLE] () => sum_9:bigint                             
                 Cost: {rows: 1 (9B), cpu: 102.00, memory: 9.00, network: 0.00} 
             - Aggregate(PARTIAL) => [sum_9:bigint]                             
                     Cost: {rows: 1 (9B), cpu: 102.00, memory: 9.00, network: 0.
                     sum_9 := "sum"("expr_3") (mask = expr_2)                   
                 - Project[] => [expr_3:bigint, expr_2:boolean]                 
                         Cost: {rows: 3 (33B), cpu: 69.00, memory: 0.00, network
                         expr_3 := (CASE "expr_2" WHEN true THEN CAST(("field" +
                     - Project[] => [expr_2:boolean, field:integer]             
                             Cost: {rows: 3 (21B), cpu: 36.00, memory: 0.00, net
                             expr_2 := (("field" + 1) > 0)                      
                         - LocalExchange[ROUND_ROBIN] () => field:integer       
                                 Cost: {rows: 3 (15B), cpu: 15.00, memory: 0.00,
                             - Values => [field:integer]                        
                                     Cost: {rows: 3 (15B), cpu: 0.00, memory: 0.
                                     (1)                                        
                                     (2)                                        
                                     (3) ``



Without the PR

``presto:sf1> explain select sum(a + 1) filter (where a + 1 > 0) from (values (1), (2), (3)) t(a);
                                          Query Plan                            
--------------------------------------------------------------------------------
 - Output[_col0] => [sum:bigint]                                                
         Cost: {rows: 1 (10B), cpu: 160.00, memory: 20.00, network: 0.00}       
         _col0 := sum                                                           
     - Aggregate(FINAL) => [sum:bigint]                                         
             Cost: {rows: 1 (10B), cpu: 160.00, memory: 20.00, network: 0.00}   
             sum := "sum"("sum_9")                                              
         - LocalExchange[SINGLE] () => sum_9:bigint                             
                 Cost: {rows: 1 (10B), cpu: 150.00, memory: 10.00, network: 0.00
             - Aggregate(PARTIAL) => [sum_9:bigint]                             
                     Cost: {rows: 1 (10B), cpu: 150.00, memory: 10.00, network: 
                     sum_9 := "sum"("expr_1") (mask = expr_2)                   
                 - Project[] => [expr_1:bigint, expr_2:boolean]                 
                         Cost: {rows: 3 (60B), cpu: 90.00, memory: 0.00, network
                         expr_1 := CAST(("field" + 1) AS bigint)                
                         expr_2 := (("field" + 1) > 0)                          
                     - LocalExchange[ROUND_ROBIN] () => field:integer           
                             Cost: {rows: 3 (30B), cpu: 30.00, memory: 0.00, net
                         - Values => [field:integer]                            
                                 Cost: {rows: 3 (30B), cpu: 0.00, memory: 0.00, 
                                 (1)                                            
                                 (2)                                            
                                 (3)   ``

Reply:
Nothing changes. `expr_3` and `expr_2` are evaluationing `x + 1`. It could due a cast. Can you please test:
```
SELECT sum(x + 1) FILTER (WHERE x + 1 > 0) FROM VALUES (BIGINT '1', BIGINT '2', BIGINT '3');
```

Reply:
``presto:sf1> explain SELECT sum(x + 1) FILTER (WHERE x + 1 > 0) FROM (VALUES (BIGINT '1'), (BIGINT '2'), (BIGINT '3')) t(x);
                                            Query Plan                          
--------------------------------------------------------------------------------
 - Output[_col0] => [sum:bigint]                                                
         Cost: {rows: 1 (9B), cpu: 135.00, memory: 18.00, network: 0.00}        
         _col0 := sum                                                           
     - Aggregate(FINAL) => [sum:bigint]                                         
             Cost: {rows: 1 (9B), cpu: 135.00, memory: 18.00, network: 0.00}    
             sum := "sum"("sum_9")                                              
         - LocalExchange[SINGLE] () => sum_9:bigint                             
                 Cost: {rows: 1 (9B), cpu: 126.00, memory: 9.00, network: 0.00} 
             - Aggregate(PARTIAL) => [sum_9:bigint]                             
                     Cost: {rows: 1 (9B), cpu: 126.00, memory: 9.00, network: 0.
                     sum_9 := "sum"("expr_3") (mask = expr_2)                   
                 - Project[] => [expr_3:bigint, expr_2:boolean]                 
                         Cost: {rows: 3 (33B), cpu: 93.00, memory: 0.00, network
                         expr_3 := (CASE "expr_2" WHEN true THEN ("field" + BIGI
                     - Project[] => [expr_2:boolean, field:bigint]              
                             Cost: {rows: 3 (33B), cpu: 60.00, memory: 0.00, net
                             expr_2 := (("field" + BIGINT '1') > BIGINT '0')    
                         - LocalExchange[ROUND_ROBIN] () => field:bigint        
                                 Cost: {rows: 3 (27B), cpu: 27.00, memory: 0.00,
                             - Values => [field:bigint]                         
                                     Cost: {rows: 3 (27B), cpu: 0.00, memory: 0.
                                     (BIGINT '1')                               
                                     (BIGINT '2')                               
                                     (BIGINT '3')     ``